File: todo/todo/api/Models/Task.cs
using System;

namespace Todo.Api.Models
{
    public class Task
    {
        public Guid Id { get; set; }
        
        public string Name { get; set; }

        public string Description { get; set; }

        public DateTime Date { get; set; }

        public Task()
        {
            Id = new Guid();
        }
    }
}

File: todo/todo/js/Task.js
class Task
{
    /**
     * 
     * @param {string} name Task name.
     * @param {string} description Task description
     * @param {Date} date Task date.
     */
    constructor(name, description, date) 
    {   
        if(typeof name !== 'string') 
            throw new ParameterTypeError('<name> must be of type <string>.');
        
        if(typeof description !== 'string') 
            throw new ParameterTypeError('<description> must be of type <string>.');

        if(!(date instanceof Date)) 
            throw new ParameterTypeError('<date> must be of type <Date>.');

        this.name = name;
        this.description = description;
        this.date = date;
        this.id;
    }
}

